src/models/set_tmp.js

```diff
@@ -0,0 +1,315 @@
+ *
+ * @see {@link set_tmp} for a version that supports scalar arguments
+ *
+ * @param {number[]} tdbArray Dry bulb air temperature, default in [°C] in [°F] if `units` = 'IP'.
+ * @param {number[]} trArray Mean radiant temperature, default in [°C]
+ * @param {number[]} vArray Air speed, default in [m/s]
+ * @param {number[]} rhArray Relative humidity, [%].
+ * @param {number[]} metArray Metabolic rate, [W/(m2)]
+ * @param {number[]} cloArray Clothing insulation, [clo]
+ * @param {number[]} wmeArray External work, [W/(m2)] default 0
+ * @param {number[]} bodySurfaceArray Body surface area, default value 1.8258 [m2] in [ft2] if units = ‘IP’
+ * @param {number[]} pAtmArray Atmospheric pressure, default value 101325 [Pa] in [atm] if units = ‘IP’
+ * @param {"standing" | "sitting"} bodyPositionArray Select either “sitting” or “standing”
+ * @param {"SI" | "IP"} [units="SI"] Select the SI (International System of Units) or the IP (Imperial Units) system.
+ * @param {boolean} [limit_inputs=true] By default, if the inputs are outsude the following limits the function returns nan. If False returns values regardless of the input values.
+ * @param {SetTmpKwargs} [kwargs]
+ * @returns {number[]} SET Array – Standard effective temperature in array, [°C]
+ *
+ * @example
+ * const set = set_tmp_array([25, 25], [25, 25], [0.1, 0.1], [50, 50], [1.2, 1.2], [0.5, 0.5]); // returns [24.3, 24.3]
+ */
+export function set_tmp_array(
+  tdbArray,
+  trArray,
+  vArray,
+  rhArray,
+  metArray,
+  cloArray,
+  wmeArray,
+  bodySurfaceArray,
+  pAtmArray,
+  bodyPositionArray,
+  units = "SI",
+  limit_inputs = true,
+  kwargs = {},
+) {
+  const defaults_kwargs = {
+    calculate_ce: false,
+    round: true,
+  };
+
+  let joint_kwargs = Object.assign(defaults_kwargs, kwargs);
+
+  if (wmeArray === undefined) {
+    wmeArray = tdbArray.map((_) => 0);
+  }
+
+  if (bodyPositionArray === undefined) {
+    bodyPositionArray = tdbArray.map((_) => "standing");
+  }
+
+  let body_surface_area = 1.8258;
+  if (bodySurfaceArray === undefined) {
+    bodySurfaceArray = tdbArray.map((_) => 1.8258);
+  }
+
+  let p_atm = 101325;
+  if (pAtmArray === undefined) {
+    pAtmArray = tdbArray.map((_) => 101325);
+  }
+
+  if (units === "IP") {
+    if (body_surface_area === 1.8258) {
+      body_surface_area = 19.65;
+    }
+    if (p_atm === 101325) {
+      p_atm = 1;
+    }
+    const unit_convert = units_converter_array(
+      {
+        tdb: tdbArray,
+        tr: trArray,
+        v: vArray,
+        area: bodySurfaceArray,
+        pressure: pAtmArray,
+      },
+      "IP",
+    );
+    tdbArray = unit_convert.tdb;
+    trArray = unit_convert.tr;
+    vArray = unit_convert.v;
+    body_surface_area = unit_convert.area;
+    p_atm = unit_convert.pressure;
+  }
+
+  let setArray = two_nodes_array(
+    tdbArray,
+    trArray,
+    vArray,
+    rhArray,
+    metArray,
+    cloArray,
+    wmeArray,
+    bodySurfaceArray,
+    pAtmArray,
+    bodyPositionArray,
+    [90],
```

Comment:
this breaks if the arrays are not of length 1

Reply:
that value doesn't do anything in `two_nodes` can you check the implementation of two_nodes pls